packages/nc-gui/components/smartsheet/toolbar/MappedBy.vue

```diff
@@ -0,0 +1,117 @@
+<script setup lang="ts">
+import type { MapType } from 'nocodb-sdk'
+import { UITypes } from 'nocodb-sdk'
+import type { SelectProps } from 'ant-design-vue'
+import {
+  ActiveViewInj,
+  IsLockedInj,
+  IsPublicInj,
+  MetaInj,
+  ReloadViewDataHookInj,
+  computed,
+  inject,
+  ref,
+  useViewColumns,
+  watch,
+} from '#imports'
+
+const meta = inject(MetaInj, ref())
+
+const activeView = inject(ActiveViewInj, ref())
+
+const IsPublic = inject(IsPublicInj, ref(false))
+
+const reloadDataHook = inject(ReloadViewDataHookInj)!
+
+const isLocked = inject(IsLockedInj, ref(false))
+
+const { fields, loadViewColumns, metaColumnById } = useViewColumns(activeView, meta, () => reloadDataHook.trigger())
+
+const { loadMapData, loadMapMeta, updateMapMeta, mapMetaData, geoDataFieldColumn } = useMapViewStoreOrThrow()
+
+const stackedByDropdown = ref(false)
+
+watch(
+  () => activeView.value?.id,
+  async (newVal, oldVal) => {
+    if (newVal !== oldVal && meta.value) {
+      await loadViewColumns()
+    }
+  },
+  { immediate: true },
+)
+
+const geoDataMappingFieldColumnId = computed({
+  get: () => mapMetaData.value.fk_geo_data_col_id,
+  set: async (val) => {
+    if (val) {
+      await updateMapMeta({
+        fk_geo_data_col_id: val,
+      })
+      await loadMapMeta()
+      await loadMapData()
+      ;(activeView.value?.view as MapType).fk_geo_data_col_id = val
+    }
+  },
+})
+
+const geoDataFieldOptions = computed<SelectProps['options']>(() => {
+  return fields.value
+    ?.filter((el) => el.fk_column_id && metaColumnById.value[el.fk_column_id].uidt === UITypes.GeoData)
+    .map((field) => {
+      return {
+        value: field.fk_column_id,
+        label: field.title,
+      }
+    })
+})
+
+const handleChange = () => {
+  stackedByDropdown.value = false
+}
+</script>
+
+<template>
+  <a-dropdown v-if="!IsPublic" v-model:visible="stackedByDropdown" :trigger="['click']">
+    <div class="nc-kanban-btn">
```

Comment:
search `kanban` in this file and replace to `map`

Reply:
done